test/verifier/universal-verifier-multi-query.test.ts

```diff
@@ -0,0 +1,501 @@
+    crossChainProofValidatorStub = await deployHelper.deployCrossChainProofValidator();
+
+    const { state } = await deployHelper.deployStateWithLibraries(["0x01A1", "0x0102"]);
+    await state.setCrossChainProofValidator(crossChainProofValidatorStub);
+    stateCrossChainStub = state;
+
+    verifier = await deployHelper.deployUniversalVerifierMultiQuery(
+      signer,
+      await stateCrossChainStub.getAddress(),
+    );
+
+    v3Validator = await deployHelper.deployValidatorStub("RequestValidatorV3Stub");
+    v3_2Validator = await deployHelper.deployValidatorStub("RequestValidatorV3_2Stub");
+    authV2Validator = await deployHelper.deployValidatorStub("AuthValidatorStub");
+
+    await verifier.addValidatorToWhitelist(await v3Validator.getAddress());
+    await verifier.addValidatorToWhitelist(await v3_2Validator.getAddress());
+    await verifier.connect();
+  }
+
+  async function checkStorageFields(verifier: any, requestId: bigint, storageFields: any[]) {
+    for (const field of storageFields) {
+      const value = await verifier.getResponseFieldValueFromAddress(
+        requestId,
+        await signer.getAddress(),
+        field.name,
+      );
+      expect(value).to.be.equal(field.value);
+    }
+  }
+
+  async function checkAuthStorageFields(verifier: any, authType: string, storageFields: any[]) {
+    for (const field of storageFields) {
+      const value = await verifier.getAuthResponseFieldValueFromAddress(
+        authType,
+        await signer.getAddress(),
+        field.name,
+      );
+      expect(value).to.be.equal(field.value);
+    }
+  }
+
+  beforeEach(async () => {
+    await loadFixture(deployContractsFixture);
+  });
+
+  it("Test submit response multiquery without groupID", async () => {
+    const requestId = 1;
+    const queryId = 1;
+    const nonExistingQueryId = 5;
+    const userId = 1;
+    const userId2 = 2;
+    const authType = "authV2";
+    const params = packV3ValidatorParams(requestQuery1);
+
+    const txSetRequests = await verifier.setRequests(
+      [
+        {
+          requestId: requestId,
+          metadata: "metadata",
+          validator: await v3Validator.getAddress(),
+          params: params,
+        },
+      ],
+      [],
+    );
+    await txSetRequests.wait();
+
+    const requestStored = await verifier.getRequest(requestId);
+    // check if the request is stored correctly checking metadata and validator
+    expect(requestStored[0]).to.be.equal("metadata");
+    expect(requestStored[1]).to.be.equal(await v3Validator.getAddress());
+    expect(requestStored[2]).to.be.equal(params);
+
+    await verifier.setAuthType({
+      authType: authType,
+      validator: await authV2Validator.getAddress(),
+      params: params,
+    });
+    const authRequestStored = await verifier.getAuthType(authType);
+    expect(authRequestStored[0]).to.be.equal(await authV2Validator.getAddress());
+    expect(authRequestStored[1]).to.be.equal(params);
```

Comment:
Consider referring to the fields like `authRequestStored.params` everywhere instead. Its more easy.

Reply:
Done